common/Cargo.toml

```diff
@@ -6,21 +6,21 @@ edition = "2018"
 license = "GPL-3.0"
 
 [dependencies]
-codec = { package = "parity-scale-codec", version = "2", default-features = false, features = ["derive", "max-encoded-len"] }
-scale-info = { version = "1.0.0", default-features = false, features = ["derive"] }
-primitive-types = { version = "0.10.1", default-features = false, features = ["scale-info"] }
+codec = { package = "parity-scale-codec", version = "3", default-features = false, features = ["derive", "max-encoded-len"] }
```

Comment:
Maybe it is better to set full version "3.1.2"?

Reply:
I think the better way use full version like in substrate: 3.0.0.   But it's not necessary

Reply:
> I think the better way use full version like in substrate: 3.0.0. But it's not necessary

These dependencies are not part of substrate, so they follow SemVer rules.